easybuild/easyblocks/p/psi.py

```diff
@@ -139,7 +140,12 @@ def configure_step(self):
         else:
             self.log.info("Using CMake based build")
             self.cfg.update('configopts', ' -DPYTHON_INTERPRETER=%s' % os.path.join(pythonroot, 'bin', 'python'))
-            self.cfg.update('configopts', ' -DCMAKE_BUILD_TYPE=Release')
+            if (self.name == 'PSI4') and (LooseVersion(self.version) >= LooseVersion("1.2")):
+                self.log.info("You have to remove CMAKE_BUILD_TYPE test in PSI4 source and the downoaded dependencies!")
+                self.log.info("Use PATCH_COMMAND in the corresponding CMakeLists.txt")
+                self.cfg.update('configopts', ' -DCMAKE_BUILD_TYPE=None')
```

Comment:
@hajgato Can you clarify this? Is this supposed to abort the build?

Reply:
@boegel 
Building `Release` version involves adding `-O3 -DNDEBUG` flags (after EB defined flags). 
There are 2 ways to avoid that, build in non-release version (then we have to patch the downloaded cmake/autocmake_safeguards.cmake)
or patch the downloaded CMakeLists.txt files to disobey `CMake` defined CMAKE_${lang}_FLAGS_RELEASE variables (Which is hardcoded in `CMake` as CMAKE_${lang}_FLAGS_RELEASE_INIT). 
Maybe there are another way, but I do not know about it or I was not thinking about it.

Reply:
@hajgato Can't we specify `-DCMAKE_CXX_FLAGS_RELEASE="$CXXFLAGS"` via `configopts`?

Reply:
@boegel 
Yes we could. But then that would apply only for `PSI4`, and not for the external projects (libint, libxc, etc, and a as a matter of fact a part of psi4 which is build as an external project), so we should patch the external project `CMakeList.txt` files for `CMAKE_CXX_FLAGS_RELEASE`. So whatever you do, you cannot solve it without patching.

Reply:
OK, but that seems better to me than using `-DCMAKE_BUILD_TYPE=None` which looks quite fishy...

How much work would it be to make sure `libxc` & co pick up on `CMAKE_CXX_FLAGS_RELEASE`?

Reply:
@boegel The problem that you have to let `CMake`  get patched the downloaded dependencies. So you have to patch the "source" CMakeLists.txt file (for each corresponding deps), then you have to make a patch for the software that will be downloaded. So in principle it is possible to do but you have to download everything to be able to do the patches (and again when you upgrade the version(s)). Maybe we can do one script that inserts the right variables to the right place in CMakeList.txt of the downloaded deps. What I feel as an advantage of the current solution that it is a simple `sed`. I know that it is an ugly hack, but in principle I could let CMake to use `-O3` everywhere, and hoping all compilation would be fine.

Just a note:
Here, I do feel that we have to find a permanent solution for `CMake`'s Release-build optimization flag hijack, as it was/will cause similar issues. I was looking here and there, but I did not find any out-of-the-box solution. It can be very annoying when developers use multiple external-projects in CMake. (like here, because in every external project you have to override the Release optimization flags, of course as long as we want to be controlled them by EasyBuild)

Reply:
@hajgato I could rename the build type from `None` to `NoAgressiveOptimizationsRelease` ...

Reply:
@hajgato Or maybe `EasyBuildRelease` or something like that, to specify that it's a custom solution for EB?

I wonder if @akesandgren has some thoughts on the CMake mess here...

Reply:
I would have to look at the actual CMake code.... Haven't had time yet though.
There are a couple of other things I need to finish up before I can spend time on this one.